server/src/server.rs

```diff
@@ -615,6 +672,56 @@ impl Geyser for GeyserService {
         Ok(resp)
     }
 
+    type SubscribeProgramUpdatesStream = SubscriptionStream<Uuid, MaybeAccountUpdate>;
+
+    async fn subscribe_program_updates(
```

Comment:
This name kind of confusing since it's __really__ just accounts that get updated and program update kind of sounds like a program deployment/upgrade. Can we just consolidate this with the existing `subscribe_account_updates` function and add an owners field to the request message to filter on

Reply:
the inspiration is [programSubscribe](https://docs.solana.com/developing/clients/jsonrpc-api#programsubscribe), so stuck with familiar terminology ever dev on solana should know.

after thinking about it a little, have a slight preference for keeping the updates separate but could be persuaded otherwise. wanna avoid overloading request filters too much right now bc building the message gets more and more complicated with different filter types. for instance, if i provide an owner field and accounts, are they or'd or and'd together? or is it a oneof filter?